### ISSUE TITLE
How to edit an UNIX timestamp?

### ISSUE THREAD

**Issue by agladysh** [json-editor/json-editor#102]
I want to represent an UNIX timestamp field as a date and time control. Is there a way to do that with json-editor?

```javascript
{
   ctime: 1528807321 // Stands for 2018-06-12T12:42:01+00:00
}
```

**Comment by pmk65** [comment]
You can extend the integer editor so that it changes the timestamp into a string for use in a "date" input field. (If you also need to be able to set the time, you could use the "datetime-local" (but it doesn't work in all browsers yet), or add an external date picker.

```javascript
JSONEditor.defaults.editors.unixdate = JSONEditor.defaults.editors.integer.extend({
    build: function () {
        this._super();
        if(!this.input) return;
        this.input.setAttribute("type", "date");

    },
    setValue: function(value) {
        if (value == "") return;
        var dateObj = new Date(value * 1000);
        this.value = this.input.value = dateObj.getFullYear() + '-' + ('0' + (dateObj.getMonth() + 1)).slice(-2) + '-' + ('0' + dateObj.getDate()).slice(-2);
    },
    getValue: function() {
        if (this.value == undefined) return;
        return (new Date(this.value).getTime() / 1000);
    }
});

JSONEditor.defaults.resolvers.unshift(function(schema) {
    if (schema.type === "integer" && schema.format === "unixdate") {
        return "unixdate";
    }
});
```

**Comment by pmk65** [comment]
This issue can be closed, as #176 enables support for dates in integer (epoch time) format.
